src/wasm/wasm-s-parser.cpp

```diff
@@ -940,6 +940,12 @@ Type SExpressionWasmBuilder::elementToType(Element& s) {
     }
     return Type(parseHeapType(*s[i]), nullable);
   }
+  if (elementStartsWith(s, RTT)) {
+    // It's an RTT, something like (rtt N $typename)
+    auto depth = atoi(s[1]->str().c_str());
```

Comment:
Thanks, I missed that. But I'm not sure what to do with it. How would we compute it? I think we could only leave it as "unspecified" or something like that. But I wonder if this is not a bug in the spec maybe?

Reply:
Oh wait, yeah, it looks like the spec even has a separate opcode for this case. Fixing...

Reply:
Looks like that constant overlaps with `exnref`. Was that removed? We still use it in binaryen it seems.

Reply:
I made it use a different constant to avoid the exnref overlap. Aside from that, it should now support rtts without a depth.

Reply:
- I haven't removed `exnref` yet, but we are not testing that in VM in Emscripten tests anymore, so it'd be better to change `exnref`'s opcode I think? Maybe add a `FIXME` to `exnref` that this is incorrect but it will be removed so it should be fine.
- Should we leave the depth unspecified in case it is not specified? Can't we do +1 to its parent's depth? I think the binary format mandates it, so every `Rtt` seems to need one; only the text format has a leeway to omit it. In the same light, I think we should not omit it when printing in the text format. Am I missing something..?

Reply:
Sorry oh wait, the binary format has a separate opcode for `rtt`s without a depth as you said...

Reply:
Sounds good, I made rtt use the right code, and made exnref use an incorrect one with a TODO to remove it. I removed one wasm binary test file that used the old exnref code.

Reply:
Sorry, nevermind what I said about +1. We can't trace all rtts up to their roots to compute it in parsing, so we should be able to parse and write rtts without depths after all. I wonder why that doesn't exist in the binary format doc though..? It's confusing which part of GC is standardized (?) and which is not...